### ISSUE TITLE
Behavior Issue during lsp loading up

### ISSUE THREAD

**Issue by YuCao16** [glepnir/lspsaga.nvim#474]
### Description
On my computer, lsp such like `pyright` and `sumneko_lua` need time to load up.
* For `sumneko_lua`, if I run `Lspsaga hover_doc` during it loading, no norify `no information available` appear, but after lsp loading, popup window then appear and won't clear when cursor moves.
* For python `lsp`, I'm using `pyright` and `pylsp` (python-lsp-server, because pyright don't support code action most of the time). I hate to say this, but I really don't konw when `Lspsaga hover_doc` would return a document hover or `no information available`. If I uninstall `pylsp`, it works fine. Is it possible to add a `lsp_blacklist` or there is another solution?

### Details

<details><summary>Reproduce</summary>

1. I have mentioned above, for `sumneko_lua`, better select a huge workspace to test :(
2. If you run `Lspsaga hover_doc` many times, then you have to navigate to popup windows and run `:q` to quit them one-by-one, and because of overlap, it looks like only one popup window appear.
</details>

<details><summary>Environment</summary>

- nvim --version output: 
```
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
```
- Operating system: `Ubuntu`

</details>

**Comment by glepnir** [comment]
oh we should add a state for some situations that server need time to deal the request.

**Comment by glepnir** [comment]
add a loading bar for hover like finder shoud be nice.

**Comment by YuCao16** [comment]
> oh we should add a state for some situations that server need time to deal the request.

Yeah, actually I'm using `fidget.nvim`, means I know when I should not run `Lspsaga hover_doc`, but sometimes still carelessly fall into the trap...

**Comment by glepnir** [comment]
also need support cancel request. if you into the trap. there will show a loadingbar that tell you hover_doc is running. then you can cancel it . easy to do .but it will be add in tomorrow. because currently I am working on neovim bug.

**Comment by YuCao16** [comment]
> also need support cancel request. if you into the trap. there will show a loadingbar that tell you hover_doc is running. then you can cancel it . easy to do .but it will be add in tomorrow. because currently I am working on neovim bug fix.

Much appreciated, is there any solution for multiple lsp config?

**Comment by glepnir** [comment]
multiple lsp config？

**Comment by glepnir** [comment]
it should work maybe something break.  it request all server s which match current filetype . I will take a look at it.

**Comment by YuCao16** [comment]
If I may ask, if there a way I can control LSP loading order if multiple client exist? 
I didn't config jedi properly, so if there is no executable python cmd in terminal, in another hand, if I don't specific python path, jedi won't be able to find documentation, but pyright can.
I checked `LspInfo` and a little neovim source code,
![Screenshot from 2022-08-30 19-32-44](https://user-images.githubusercontent.com/62466929/187516368-1537a841-a877-4a05-affd-8b219625d80a.png)
If jedi is on top, then run `Lspsaga hover_doc` it will only show `no information available`, and if pyright is on top, it will work just fine. So in this case, if one could set two LSP loading order, it would be more stable.

**Comment by glepnir** [comment]
I think no matter the order. hover_doc should be work. it will request all servers in buffer . and I see someone use pylsp and pyright. maybe this is better?

**Comment by YuCao16** [comment]
> I think no matter the order. hover_doc should be work. it will request all servers in buffer . and I see someone use pylsp and pyright. maybe this is better?

Actually I'm not sure if this worth time to solve, but here is the case:

The reason I still use jedi is because it provide better document hover, it has code highlight and helpful signature help (In my opinion much better than pyright). But other things jedi is not good at, like jedi reports only syntax error, so I use jedi and pyright at the same time. Pylsp is another solution but I don't know why its document hover is the same as pyright, I would double check.

So in short, if lspsaga could receive documents from ordered lsp client, things would be a lot easier. I think this would be a meaningful change for any user who uses multiple lsp client.

PS, for neovim, when using two lsps, isn't it strange that each time they are loaded in a different order and makes the results different, or is there a really very small population that uses multiple lsp clients?

**Comment by glepnir** [comment]
the multiples lsp clients often appear in front language or project. I mean there had something wrong in lspsaga hover.

so maybe the first lsp return a empty result. there I simple check if `res.result` then break. so bug in here maybe .need some test.  

https://github.com/glepnir/lspsaga.nvim/blob/57fd6cfc25dbb02e041605e500a55648aea58e01/lua/lspsaga/hover.lua#L113-L116

**Comment by glepnir** [comment]
@YuCao16  can you post your server config here

**Comment by YuCao16** [comment]
Sorry, behavior seems become more strange, I created a small record, if I call `Lspsaga hover_doc` for a function (my_func), it will return document hover for `return` command, and a popup window appear and disappear. 
![demo](https://user-images.githubusercontent.com/62466929/187786383-d20d089c-ba26-48a9-bf4c-7e39377babdb.gif)

**Comment by YuCao16** [comment]
> @YuCao16 can you post your server config here
Sorry for the delay, I shared my config [here](https://github.com/YuCao16/dotfiles/tree/main/dot_config/nvim/lua/configs/lsp).

**Comment by YuCao16** [comment]
Hello,
Similar problem happens on `Lspsaga lsp_finder`.
When I using `clangd` with big package such as libtorch, if I move cursor to type alias such as `int64_t`or `size_t`, loadingbar will stuck at (`'█▁▁▁▁▁▁▁▁▁',`) and took several seconds to load. I have set `finder_request_timeout = 2000` but doesn't help.

**Comment by glepnir** [comment]
kitty ?

**Comment by YuCao16** [comment]
> kitty ?

Yes

**Comment by glepnir** [comment]
the animate of loading bar in Kitty will hanged in first time open finder. in wezterm is fine. all lsp request are async. so we want await all response from server. make it silent looks weird. because you don't know what happened after call finder. increase the request time out then did you get the results? finder window not appear?

**Comment by YuCao16** [comment]
> the animate of loading bar in Kitty will hanged in first time open finder. in wezterm is fine. all lsp request are async. so we want await all response from server. make it silent looks weird. because you don't know what happened after call finder. increase the request time out then did you get the results? finder window not appear?

If I'm patient enough, the finder window will appear. But if I switch windows(<c-w>*2) in this process, then the find preview window will stay, but finder window won't.

**Comment by glepnir** [comment]
> But if I switch windows(*2) in this process, then the find preview window will stay, but finder window won't.

this maybe a bug please create an new issue for it
